packages/service-core/test/src/auth.test.ts

```diff
@@ -331,12 +342,34 @@ describe('JWT Auth', () => {
 
     currentResponse = Promise.resolve({
       errors: [],
-      keys: [await KeySpec.importKey(publicKey)]
+      keys: [await KeySpec.importKey(publicKeyRSA)]
     });
 
     // After a delay, we can refresh again
     await cached.addTimeForTests(30_000);
     key = (await cached.getKeys()).keys[0];
-    expect(key.kid).toEqual(publicKey.kid!);
+    expect(key.kid).toEqual(publicKeyRSA.kid!);
+  });
+
+  test('signing with EdDSA', async () => {
+    const keys = await StaticKeyCollector.importKeys([publicKeyEdDSA]);
+    const store = new KeyStore(keys);
+    const signKey = (await jose.importJWK(publicKeyEdDSA)) as jose.KeyLike;
+
+    const signedJwt = await new jose.SignJWT({ claim: 'test-claim' })
```

Comment:
The test is failing here - it currently attempts to sign using the _public_ key, while it should sign using the _private_ key.

Reply:
Thanks! Looks good now. I'll test properly next week, then we can merge.